teos/src/main.rs

```diff
@@ -311,13 +326,17 @@ async fn main() {
 
     // The ordering here actually matters. Listeners are called by order, and we want the gatekeeper to be called
     // first so it updates the users' states and both the Watcher and the Responder operate only on registered users.
-    let listener = &(gatekeeper, &(watcher.clone(), responder));
+    let listeners = (gatekeeper, (watcher.clone(), responder));
+
+    // This spawns a separate async actor in the background that will be fed new blocks from a sync block listener.
+    // In this way we can have our components listen to blocks in an async manner from the async actor.
+    let listener = AsyncBlockListener::wrap_listener(listeners, dbm);
+
     let cache = &mut UnboundedCache::new();
-    let spv_client = SpvClient::new(tip, poller, cache, listener);
+    let spv_client = SpvClient::new(tip, poller, cache, &listener);
     let mut chain_monitor = ChainMonitor::new(
```

Comment:
I think it's gonna be cleaner to merge all of these into the chain monitor.
And also merge `async_listener.rs` into `chain_monitor.rs` and change the chain monitor tests to accommodate the `AsyncListen` trait.